### ISSUE TITLE
https proxifying through a http proxy

### ISSUE THREAD

**Issue by rezad1393** [semigodking/redsocks#83]
I am trying to use transparent on openwrt and then chain redsocks with pacproxy app (https://github.com/williambailey/pacproxy) which is a http proxy.

when I use (export https_proxy=192.168.1.1:8080) from my laptop and then use from command line use "curl https://ping.eu" the pacproxy app works and in it logs I get 

"2018/02/26 19:39:08.561046 proxyhttphandler.go:82: Proxy Lookup "//ping.eu:443", got "DIRECT". Selected "DIRECT""
but when when I use the empty proxy so that the router proxify the connections then for"curl https://ping.eu"  I get this:

"2018/02/26 19:42:50.475683 proxyhttphandler.go:82: Proxy Lookup "//88.198.46.60:443", got "DIRECT". Selected "DIRECT""

so the pacpoxy gets ip of from redsocks when I use http-connect or http-relay.
so it cant use pac proxy file to selects the correct proxy.

I cant use socks5(which works I think for other proxies,like tor, that use domains not IPs) with pacproxy, because it supports only http proxy, but it works with proxifying https connections on my on laptop using export .

also I tried to use https-connect for type in redsocks2 config but it gives error that is not correct type.

so to sum it up if I use https_proxy envirrement variable from my laptop to use the pacprxoy app on the router and then making a connection to a https site, then pacproxy log shows that it correctly saw the domain and decides based on the pac file provided to what downstream proxy to use to cooecnt to that site.
but if I use the redsocks to redirect https connections to the pacproxy proxy , then in its log it doesnt see domain names but ip addresses of the domians and that messes up the decision for selecting the correct downstream proxy for that domain.

my description is a bit confusing so if there is any question please ask them.

**Comment by semigodking** [comment]
It is not an easy job for any transparent proxy to get accurate host name for redirected connections. See #69 for workaround.

**Comment by rezad1393** [comment]
why http one works then? just https proxy doesnt work (shows ips instead of domains)

**Comment by semigodking** [comment]
As you mentioned, for https traffic, we need to extract domain information from SNI. redsocks does not support this. For http traffic, domain name is extracted from HTTP header. 
You might want to have a look at [this project](https://github.com/shadowsocks/shadowsocks-libev) which has SNI support.

**Comment by rezad1393** [comment]
so that project replaces resock2 or does it replace the pacproxy in my case?

**Comment by semigodking** [comment]
No, it is not a replacement.  It is mainly focus on censorship circumstance
and has very good ecosystem in popular OS's.

**Comment by rezad1393** [comment]
I dont understand your answer.
it uses iptables rules just like redsocks2. so do I redirect the connection to that app or to redsocks them from redsocks to that app.

**Comment by semigodking** [comment]
it has its own protocol, server side app and client side apps. Just like a
socks server and client. It has a client side tool which can transparent
proxy HTTPS traffic to its server side via its protocol.

**Comment by rezad1393** [comment]
so the client gets the sni info from the packets?
I search a lot for the pac option but it is all in chinese or about servers outside of local network. so it is useless to me.

does redsocks have any plan for sni support?

**Comment by semigodking** [comment]
unfortunately, no.
